### DISCUSSION TITLE
Moving multiple items into a Collection

### DISCUSSION THREAD

**Discussion by adamshand** [dani-garcia/vaultwarden#2705] [Q&A] [answered]
I've just imported an old Keepass file into Vaultwarden Organisation. I am now wanting to move all of these items into a Collection.

I can see that there is a way to move multiple items into a folder, but I can't see any way to do this with Collections? I can only see a way to move items one by one. :-(

I assume that I'm missing something obvious? Many thanks for any pointers.

**Comment by RealOrangeOne** [comment, marked as answer]
You aren't i'm afraid - it's just not possible currently.

Worth noting this is a Bitwarden limitation, not Vaultwarden's.

See also https://community.bitwarden.com/t/bulk-collection-migration/14489/2

**Reply by adamshand** [reply]
Stink! Good to see that it's under development in Bitwarden, thanks for the link!

**Comment by 0xquad** [comment]
Landed here after a web search. Bitwarden still haven't implemented anything to resolve this yet. Someone in their community forum posted a solution using the Bitwarden CLI (`bw`) ~~but it's not applicable To Vaultwarden~~ (edit: the BW CLI can apparently be used for Vaultwarden too).

I came up with a piece of Javascript to reassign a bunch of items to a collection instead. Run this in the console of a web session to Vaultwarden.

https://gist.github.com/0xquad/89077b1245425f17f4476336b0701ef4

You need to gather some info and plug the values in the script before running it.

**Reply by BlackDex** [reply]
Why isn't the cli solution applicable for Vaultwarden?

**Reply by 0xquad** [reply]
Good point; actually I just assumed that since there was no CLI for Vaultwarden (right?), it wouldn't apply, but one can always install the Bitwarden CLI from I've just seen (but haven't really read about it). Never tried that thought, so I'm not sure how complex it could become, and I'm wondering if the Javascript snippet isn't just easier in any case. Feel free to chip in with more info about that!

**Reply by BlackDex** [reply]
It's very easy. Vaultwarden is compatible with the Bitwarden clients.

https://bitwarden.com/help/change-client-environment/#tab-cli-4dQ4hW1QAwVBuReXk2Txx0

**Comment by iviaxpow3r** [comment]
Bitwarden has added this feature at this point (As of March 2024) - https://github.com/bitwarden/clients/releases/tag/web-v2024.3.1

is it possible to get the concept integrated into Vault Warden?

**Reply by BlackDex** [reply]
That is possible.

**Reply by ac-julian** [reply]
That would be amazing!

**Reply by FrankyBoy** [reply]
Colleague just ran into the same issue when trying to restructure our data. moving in bulk would be highly appreciated

**Reply by thumDer** [reply]
Are there any plans to include it in the near future? It would be highly appreciated, yes!

**Reply by BlackDex** [reply]
Once we can update the web-vault to a newer version, then yes. The backend already supports it though.
Only the web-vault is currently holding us back.

**Reply by JFQueralt** [reply]
Has there been any updates on this?
